frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -998,7 +999,9 @@ export default ({
               if (!fromOurselves && isScrollable) {
                 this.updateScroll()
               } else {
-                const msg = this.messages.filter(m => !m.pending).slice(-1)
+                // If there are any temporary messages that do not exist in the
+                // contract, they should not used for updateReadUntilMessageHash
+                const msg = this.messages.filter(m => !m.pending && !m.hasFailed).pop()
```

Comment:
`.pop()` is safe to use because it's done on the result of `.filter()`, which is a new array. Since this looking at `this.messages` in general, there could be pending or failed messages there that need to be excluded.